cache/src/main/java/io/envoyproxy/controlplane/cache/SimpleCache.java

```diff
@@ -363,4 +564,89 @@ private boolean respond(Watch watch, U snapshot, T group) {
 
     return false;
   }
+
+  private List<String> findRemovedResources(DeltaWatch watch, Map<String, VersionedResource<?>> snapshotResources) {
+    // remove resources for which client has a tracked version but do not exist in snapshot
+    return watch.trackedResources().keySet()
+        .stream()
+        .filter(s -> !snapshotResources.containsKey(s))
+        .collect(Collectors.toList());
+  }
+
+  private Map<String, VersionedResource<?>> findChangedResources(DeltaWatch watch,
+      Map<String, VersionedResource<?>> snapshotResources) {
+    return snapshotResources.entrySet()
+        .stream()
+        .filter(entry -> {
+          if (watch.pendingResources().contains(entry.getKey())) {
+            return true;
+          }
+          String resourceVersion = watch.trackedResources().get(entry.getKey());
+          if (resourceVersion == null) {
+            // resource is not tracked, should respond it only if watch is wildcard
+            return watch.isWildcard();
+          }
+          return !entry.getValue().version().equals(resourceVersion);
+        })
+        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+  }
+
+  private ResponseState respondDeltaTracked(DeltaWatch watch,
```

Comment:
@mikegajda is this used somewhere?

Reply:
Hmmm, I don't see any usages either, so I think it's safe to remove